specs/Pills.md

```diff
@@ -0,0 +1,188 @@
+
+  /**
+   * A button shorthand for the dismiss action slot. To use this slot the pill should be
+   * dismissible.
+   */
+  dismissAction?: ShorthandValue<AlertDismissActionProps>;
+}
+```
+
+## Structure
+
+- _**Public**_
+
+```jsx
+<Pills>
+  <Pill />
+</Pills>
+```
+
+- _**Internal**_
+
+```jsx
+<ElementType >
+  <li>
+    <span>
+      {children}
+    </span>
+  <li>
+</ElementType>
+```
+
+- _**DOM**_
+
+```html
+<ul aria-label>
+  <li role="presentation">
+    <span role="button">
+      ...
+    </span>
+  </li>
+</ul>
+```
+
+## Migration
+
+- _Migration from v8_
+
+...
+
+- _Migration from v0_
+
+`Dropdown` component should use `Pills` over `DropdownSelectedItem`.
+
+### Pill with dismiss
+
+#### Accessibility
+
+- Each pill is of role="button"
+- Tab key to set focus on the Pill
+- Tab key moves focus to next item
+- Esc key remove focus
+
+#### Keyboard Navigation
+
+| key                | state    | title                                                              |
+| ------------------ | -------- | ------------------------------------------------------------------ |
+| left/right         | OnFocus  | Role= “button”, aria-describedby=”Press enter or delete to remove” |
+| Enter/Delete       | OnDelete | The pill is removed since there is only one primary action         |
+| Backspace/Ctrl + X | OnDelete | Backspace/Ctrl + X                                                 |
+
+### Pill with popup
+
+#### Accessibility
+
+- Each pill is of role="button"
+- Popup opens after 500ms of mouse hover state
+- Clicking in the Pill open popup and clicking outside should dimiss it
+
+#### Keyboard Navigation
+
+| key                | state    | title                                                              |
+| ------------------ | -------- | ------------------------------------------------------------------ |
+| Esc                | OnFocus  | Dismis popup                                                       |
+| Enter              | OnFocus  | Open Popup                                                         |
+| Backspace/Ctrl + X | OnDelete | Backspace/Ctrl + X                                                 |
+| left/right         | OnFocus  | Role= “button”, aria-describedby=”Press enter or delete to remove” |
+
+### Pill with right click (invoking context menu)
+
+#### Accessibility
+
+- Each pill is of role="button"
+- Right click anywhere on card surface invokes content menu. Context menu appears close to cursor location on card
+- Clicking or tapping anywhere outside of menu to dismiss
+
+#### Keyboard Navigation
+
+| key       | state    | title                          |
+| --------- | -------- | ------------------------------ |
+| Esc       | OnFocus  | Dismis context menu            |
+| Shift+F10 | OnFocus  | Open context menu              |
+| up/down   | Expanded | Navigates context menu options |
+
+### Pill with toggle
+
+#### Accessibility
+
+- Each pill is of role="button"
+- Click toggles Pill's selected state
```

Comment:
Do we need different aria-attributes or roles for this ? 

i.e. aria-selected/role="option" or if we stick with button then aria-pressed ?

Reply:
DOM out put for each of the variations would be nice if they differ from the default

Reply:
I would go with `role="option"` I set as `button` as per design spec but don't think it's the best